jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerCredentialHelper.java

```diff
@@ -54,30 +54,30 @@
    * Construct with {@link DockerCredentialHelperFactory}.
    *
    * @param serverUrl the server URL to pass into the credential helper
-   * @param credentialHelperSuffix the credential helper CLI suffix
+   * @param credentialHelper the path to the credential helper executable
    */
-  DockerCredentialHelper(String serverUrl, String credentialHelperSuffix) {
+  DockerCredentialHelper(String serverUrl, Path credentialHelper) {
     this.serverUrl = serverUrl;
-    this.credentialHelperSuffix = credentialHelperSuffix;
+    this.credentialHelper = credentialHelper;
   }
 
   /**
-   * @return the Docker credentials by calling the corresponding CLI.
-   *     <p>The credential helper CLI is called in the form:
-   *     <pre>{@code
+   * Calls the credential helper CLI in the form:
+   *
+   * <pre>{@code
    * echo -n <server URL> | docker-credential-<credential helper suffix> get
    * }</pre>
    *
+   * @return the Docker credentials by calling the corresponding CLI
    * @throws IOException if writing/reading process input/output fails.
    * @throws NonexistentServerUrlDockerCredentialHelperException if credentials are not found.
    * @throws NonexistentDockerCredentialHelperException if the credential helper CLI doesn't exist.
    */
-  public Authorization retrieve()
+  public Credentials retrieve()
```

Comment:
Can't DockerCredentialHelper be a `CredentialsRetriever`?

Reply:
Yep, we can definitely make it an implementation of `CredentialsRetriever`. I added it as a task to do after we remove the uses in `RetrieveRegistryCredentialsStep`.